wordpress-mobile/WordPress-Android
# Issue/13328 restore other running

Parent #13328 

This PR updates the GetRestoreUseCase to check for a currently running restore before posting a new restore request.

To test:
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog 
- Tap the more menu (note the date of the item)
- Select the restore option
- Tap the Restore Site button on the Details View
- Leave the app open on the confirm view
- Log on to the web
- Navigate to activity log and kick off a restore for a different date
- Go back to the app & tap the confirm restore button
- Note that you go back to the details view and are shown a snackbar for other request running

PR submission checklist:
